File: product-service/src/main/java/com/github/buyalsky/productservice/dto/ProductPriceChangeDto.java
package com.github.buyalsky.productservice.dto;

import java.util.Objects;

public class ProductPriceChangeDto {
    private String productId;
    private Double previousPrice;
    private Double newPrice;

    public ProductPriceChangeDto() {
    }

    public ProductPriceChangeDto(String productId, Double previousPrice, Double newPrice) {
        this.productId = productId;
        this.previousPrice = previousPrice;
        this.newPrice = newPrice;
    }

    public String getProductId() {
        return productId;
    }

    public Double getPreviousPrice() {
        return previousPrice;
    }

    public Double getNewPrice() {
        return newPrice;
    }

    public void setProductId(String productId) {
        this.productId = productId;
    }

    public void setPreviousPrice(Double previousPrice) {
        this.previousPrice = previousPrice;
    }

    public void setNewPrice(Double newPrice) {
        this.newPrice = newPrice;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        ProductPriceChangeDto that = (ProductPriceChangeDto) o;
        return productId.equals(that.productId) && previousPrice.equals(that.previousPrice) && newPrice.equals(that.newPrice);
    }

    @Override
    public int hashCode() {
        return Objects.hash(productId, previousPrice, newPrice);
    }
}

File: shopping-cart-service/src/main/kotlin/com/github/buyalsky/shoppingcartservice/dto/ProductPriceChangeDto.kt
package com.github.buyalsky.shoppingcartservice.dto

import com.github.buyalsky.shoppingcartservice.dto.ProductPriceChangeDto
import java.util.*

class ProductPriceChangeDto {
    var productId: String? = null
    var previousPrice: Double? = null
    var newPrice: Double? = null
    var productName: String? = null

    constructor()

    constructor(productId: String?, productName: String?, previousPrice: Double?, newPrice: Double?) {
        this.productId = productId
        this.productName = productName
        this.previousPrice = previousPrice
        this.newPrice = newPrice
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val that = o as ProductPriceChangeDto
        return productId == that.productId && previousPrice == that.previousPrice && newPrice == that.newPrice
                && productName == that.productName
    }

    override fun hashCode(): Int {
        return Objects.hash(productId, productName, previousPrice, newPrice)
    }
}
